src/DotNet.Testcontainers/Containers/Modules/Databases/CouchbaseTestcontainer.cs

```diff
@@ -0,0 +1,39 @@
+namespace DotNet.Testcontainers.Containers.Modules.Databases
+{
+  using System.Linq;
+  using System.Threading.Tasks;
+  using Abstractions;
+  using Configurations;
+
+  public class CouchbaseTestcontainer : TestcontainerDatabase
+  {
+    internal CouchbaseTestcontainer(ITestcontainersConfiguration configuration) : base(configuration)
+    {
+    }
+
+    public override string ConnectionString => "couchbase://" + this.Hostname;
+
+    public async Task<long> CreateBucket(string bucket, int memory = 512, int withDelaySecond = 3)
+    {
+      var createBucketCommand =
+        "bash /opt/couchbase/bin/couchbase-cli " +
+        "bucket-create -c 127.0.0.1:8091 " +
+        "--username " + this.Username + " --password " + this.Password + " " +
+        "--bucket=" + bucket + " --bucket-type couchbase --bucket-ramsize "+memory+" --enable-flush 1 --bucket-replica 0";
+      await Task.Delay(withDelaySecond * 1000);
```

Comment:
Why do we need this delay? I think we should remove it (`FlushBucket` too).

Reply:
Execution to these commands over Couchbase taking a bit time because of Couchbase index mechanism. It's totally depending to resources. In local environment you can see results without any delay. But depending CI/CI pipeline environment resources the delay may need to use. On the other hand, this is not a responsibility of library. Library user can consider this situation depending ci/ci pipeline. What is your suggestion? Should we remove?

Reply:
Well I think that call should be executed after `this.ExecAsync()` (if at all) and then we should use `Task.DelayTimeSpan.FromSeconds(withDelaySecond))`.

However, getting rid of it would be the best. Can we check with `couchbase-cli` if the bucket exist? Then we can use the behaviour of the wait strategies. Such as:

```csharp
public Task<bool> DoesBucketExists(string bucketName, CancellationToken ct = default)
{
  return Task.FromResult(true); // Add couchbase-cli call here and check exit code or something like that.
}

public async Task<long> CreateBucket(string bucketName, int memory = 512, int timeout = 1000, CancellationToken ct = default)
{
  const string createBucketCommand = "couchbase-cli create bucket command";
  var exitCode = await this.ExecAsync(new[] {"/bin/sh", "-c", createBucketCommand  }, ct);

  await WaitStrategy.WaitUntil(() => this.DoesBucketExists(bucketName, ct), timeout: timeout, ct: ct);
  return exitCode;
}
```